Mathlib/RingTheory/NoetherNormalization.lean

```diff
@@ -0,0 +1,287 @@
+/-
+Copyright (c) 2024 Sihan Su. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca, Sihan Su, Wan Lin, Xiaoyang Su
+-/
+import Mathlib.Algebra.MvPolynomial.Monad
+import Mathlib.Data.List.Indexes
+import Mathlib.RingTheory.IntegralClosure.IsIntegralClosure.Basic
+/-!
+# Noether Normalizaton
+This file contains a proof of Noether normalization lemma by Nagata.
+
+## Main Results
+Let $A$ be a finitely generated algebra over a field $k$.
+Then there exists natural number $r$ and an injective homomorphism
+from $k[X_1, X_2, ..., X_r]$ to $A$ such that $A$ is integral over $k[X_1, X_2, ..., X_r]$.
+
+## Reference
+* <https://stacks.math.columbia.edu/tag/00OW>
+
+-/
+open Polynomial MvPolynomial Ideal BigOperators Nat RingHom
+
+variable {k : Type*} [Field k] {n : ℕ} (f : MvPolynomial (Fin (n + 1)) k)
+variable (v w : Fin (n + 1) →₀ ℕ)
+
+/-- Suppose $f$ is a nonzero polynomial of $n+1$ variables : $X_0,...,X_n$.
+`up` is a number which is big enough. -/
+noncomputable abbrev up := 2 + f.totalDegree
```

Comment:
```suggestion
local notation3 "up" => 2 + f.totalDegree
```
It seems better to have it as a simple notation (so it is completely transparent to Lean). Note that you have to remove `f` from `up f` everywhere.